airflow/api_fastapi/execution_api/routes/assets.py

```diff
@@ -0,0 +1,79 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Annotated
+
+from fastapi import HTTPException, Query, status
+from sqlalchemy import select
+
+from airflow.api_fastapi.common.db.common import SessionDep
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.execution_api.datamodels.asset import AssetResponse
+from airflow.models.asset import AssetModel
+
+# TODO: Add dependency on JWT token
+router = AirflowRouter(
+    responses={
+        status.HTTP_404_NOT_FOUND: {"description": "Asset not found"},
+        status.HTTP_401_UNAUTHORIZED: {"description": "Unauthorized"},
+    },
+)
+
+
+@router.get(
+    "",
+    responses={
+        status.HTTP_400_BAD_REQUEST: {
+            "description": "Either 'name' or 'uri' query parameter must be provided"
+        },
```

Comment:
I don't love this (just because of how messy it makes this endpoint's code). What about using two endpoints `assets//by-name` and `assets/by-uri/` as suggested here https://stackoverflow.com/a/30551711/439189 

Thoughts? Not a strong feeling on my part, just this feels "a little bit messy"

Reply:
That is exactly how I had it earlier :) -- and for a similar reason: I wanted to enforce validation at the API layer itself. (https://stackoverflow.com/a/76044727/5691525)

I think I changed it coz there was some code duplication on test side --- but I agree that separate endpoint is better, I will change it